WordPress/Classes/Services/PostRepository.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+final class PostRepository {
+
+    enum Error: Swift.Error {
+        case postNotFound
+        case remoteAPIUnavailable
+    }
+
+    private let coreDataStack: CoreDataStackSwift
+    private let remoteFactory: PostServiceRemoteFactory
+
+    init(coreDataStack: CoreDataStackSwift, remoteFactory: PostServiceRemoteFactory = PostServiceRemoteFactory()) {
+        self.coreDataStack = coreDataStack
+        self.remoteFactory = remoteFactory
+    }
+
+    /// Sync a specific post from the API
+    ///
+    /// - Parameters:
+    ///   - postID: The ID of the post to sync
+    ///   - blogID: The blog that has the post.
+    /// - Returns: The stored post object id.
+    func getPost(withID postID: NSNumber, from blogID: TaggedManagedObjectID<Blog>) async throws -> TaggedManagedObjectID<AbstractPost> {
+        let remote = try await coreDataStack.performQuery { [remoteFactory] context in
+            let blog = try context.existingObject(with: blogID)
+            return remoteFactory.forBlog(blog)
+        }
+
+        guard let remote else {
+            throw PostRepository.Error.remoteAPIUnavailable
+        }
+
+        let remotePost: RemotePost? = try await withCheckedThrowingContinuation { continuation in
+            remote.getPostWithID(
+                postID,
+                success: { continuation.resume(returning: $0) },
+                failure: { continuation.resume(throwing: $0!) }
```

Comment:
For future reference, I drilled down the call stacks and I _think_ this force unwrap should always be safe. 

I ended up with calls into `WordPressComRestApi` that always have a concrete `NSError` value.

- https://github.com/wordpress-mobile/WordPressKit-iOS/blob/trunk/WordPressKit/WordPressComRestApi.swift#L207-L209
- https://github.com/wordpress-mobile/WordPressKit-iOS/blob/8955556d4ca410940ddabbea7be56043f17f6616/WordPressKit/WordPressComRestApi.swift#L544

Reply:
Also, this `remote.getPostWithID` function [calls `WordPressComRestApi.GET`](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/8.5.1/WordPressKit/PostServiceRemoteREST.m#L43) which [declares its error argument as non-null](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/8.5.1/WordPressKit/WordPressComRestApi.swift#L54). That means we are getting an error argument with Optional type due to a function declaration issue, not an implementation issue: `PostSerivceRemote` should have declared the error as non-null.